[WIP] Add "Not Schedulable" to NodeStatus on nodes list page

benjaminapetersen opened the pull request:
Only WIP because I have not been able to test locally.  My dev console currently will not run, but I'll verify this soon as possible tomorrow.

Fixes [Bugzilla 1608742](https://bugzilla.redhat.com/show_bug.cgi?id=1608742)

spadgett commented on frontend/public/components/node.tsx:
```diff
@@ -54,7 +54,15 @@ const HeaderSearch = props => <ListHeader>
+  if (isNodeReady(node)) {
+    return <span className="node-ready"><i className="fa fa-check"></i> Ready</span>;
+  }
+  if (!isNodeSchedulable(node)) {
```
It looks like it will short-circuit before it gets here if the node is ready, but not schedulable.

benjaminapetersen replied:
Prob the latter seems sufficient?

benjaminapetersen commented:
Ok, updated to:

![screen shot 2018-07-27 at 12 06 36 pm](https://user-images.githubusercontent.com/280512/43332279-e65b4d0a-9195-11e8-8c09-774a7518af64.png)
![screen shot 2018-07-27 at 12 06 51 pm](https://user-images.githubusercontent.com/280512/43332280-e6672a26-9195-11e8-8a95-7218fd8b0bf6.png)

Open to different colors/icons/label text if anyone has a strong pref.
@tlwu2013 @rhamilto @sg00dwin

benjaminapetersen commented:
CLI shows both `Ready, Scheduling Disabled`:

![screen shot 2018-07-27 at 3 51 00 pm](https://user-images.githubusercontent.com/280512/43343661-04d65c1a-91b5-11e8-945f-270bffd9c506.png)

This is accurate so is probably better.  It just looks a little ugly with the overflow.

spadgett commented:
If you want to see the CLI logic, it's here:

https://github.com/kubernetes/kubernetes/blob/v1.12.0-alpha.0/pkg/printers/internalversion/printers.go#L1106

rhamilto commented:
>  It just looks a little ugly with the overflow.

Overflow is fixable.  See https://github.com/openshift/console/blob/3043355b3c8eca5e6f28dca454c75793b12da404/frontend/public/components/_icon-and-text.scss

spadgett commented:
If we're going to show both, we might as well stack them since it will wrap anyway.

I'd probably just take the hourglass icon off scheduling disabled. I'm not sure it fits (although I know we use that for pending routes, etc.)

benjaminapetersen commented:
I can leave "✔ready", and put "scheduling disabled" below (stacked)

benjaminapetersen commented:
Rebased, updated to stacked without an icon, doesn't look fantastic:

![screen shot 2018-07-31 at 3 50 57 pm](https://user-images.githubusercontent.com/280512/43483415-9a1176d0-94d9-11e8-80ea-61729e545386.png)

But it might be fine as this table isn't super responsive yet anyway:

![screen shot 2018-07-31 at 3 52 11 pm](https://user-images.githubusercontent.com/280512/43483460-bfd0c0f6-94d9-11e8-9ad8-381ac9c8b0a8.png)

rhamilto commented on frontend/public/components/node.tsx:
```diff
@@ -54,7 +54,22 @@ const HeaderSearch = props => <ListHeader>
+    if (!isNodeSchedulable(node)) {
+      return <React.Fragment>
+        <span className="node-ready co-icon-and-text"><i className="fa fa-check co-icon-and-text__icon" aria-hidden="true"></i> Ready</span>
+        <span className="text-muted co-icon-and-text">Scheduling Disabled</span>
```
Semantically this isn't an icon and text.

spadgett commented:
/close

Closing since we've recently reworked the nodes list for machines.
